llvm/lib/Transforms/InstCombine/InstCombineCasts.cpp

```diff
@@ -103,6 +103,13 @@ Value *InstCombinerImpl::EvaluateInDifferentType(Value *V, Type *Ty,
       }
     }
     break;
+  case Instruction::ShuffleVector: {
+    Value *Op0 = EvaluateInDifferentType(I->getOperand(0), Ty, isSigned);
+    Value *Op1 = EvaluateInDifferentType(I->getOperand(1), Ty, isSigned);
```

Comment:
Ooops, this is not correct if the shuffle operand types are not the same as the result type. In that case we need to adjust to a different type with the same element type but different number of elements.

Reply:
Can I revert this in the meantime? (unless you already have a fix for this)

Reply:
I reverted this. Feel free to reland after the fix.